src/test/java/hudson/plugins/emailext/MailAccountFIPSTest.java

```diff
@@ -0,0 +1,115 @@
+        account.setAddress("joe@example.com");
+        assertTrue(account.isValid());
+    }
+
+    @Test
+    @WithoutJenkins
+    public void testIsNotValidAuthConfigWithoutEncryption() {
+        MailAccount account = new MailAccount();
+        account.setAddress("joe@example.com");
+        account.setCredentialsId("foo");
+        assertFalse(account.isValid());
+    }
+
+    @Test
+    @WithoutJenkins
+    public void testIsValidAuthConfigAndSSL() {
+        MailAccount account = new MailAccount();
+        account.setAddress("joe@example.com");
+        account.setCredentialsId("foo");
+        account.setUseSsl(true);
+        assertTrue(account.isValid());
+    }
+
+    @Test
+    @WithoutJenkins
+    public void testIsValidAuthConfigAndTLS() {
+        MailAccount account = new MailAccount();
+        account.setAddress("joe@example.com");
+        account.setCredentialsId("foo");
+        account.setUseTls(true);
+        assertTrue(account.isValid());
+    }
+
+    @Test
+    public void testFormValidationForInsecureAuth() {
+        final String validCredentialId = "valid-id";
+        SystemCredentialsProvider.getInstance()
+                .getCredentials()
+                .add(new UsernamePasswordCredentialsImpl(
+                        CredentialsScope.GLOBAL, validCredentialId, "description", "username", "password"));
+
+        MailAccountDescriptor mad = (MailAccountDescriptor) Jenkins.get().getDescriptor(MailAccount.class);
+
+        assertThat(mad.doCheckCredentialsId(null, "", false, false), hasKind(Kind.OK));
+        assertThat(mad.doCheckCredentialsId(null, null, false, false), hasKind(Kind.OK));
+
+        // no auth but any combination of TLS/SSL is ok
+        assertThat(mad.doCheckCredentialsId(null, null, true, false), hasKind(Kind.OK));
+        assertThat(mad.doCheckCredentialsId(null, null, false, true), hasKind(Kind.OK));
+        assertThat(mad.doCheckCredentialsId(null, null, true, true), hasKind(Kind.OK));
+
+        // valid credentials with TLS
+        assertThat(mad.doCheckCredentialsId(null, validCredentialId, true, false), hasKind(Kind.OK));
+        assertThat(mad.doCheckCredentialsId(null, validCredentialId, false, true), hasKind(Kind.OK));
+        assertThat(mad.doCheckCredentialsId(null, validCredentialId, true, true), hasKind(Kind.OK));
+
+        // valid credentials without TLS produce a error
+        assertThat(
+                mad.doCheckCredentialsId(null, validCredentialId, false, false),
+                allOf(hasKind(Kind.ERROR), hasMessage("Authentication requires either TLS or SSL to be enabled")));
+
+        // non-valid creds show the error regardless of SSL/TLS
+        assertThat(
+                mad.doCheckCredentialsId(null, "bogus", false, false),
+                allOf(
+                        hasKind(Kind.ERROR),
+                        hasMessage(containsString("Authentication requires either TLS or SSL to be enabled")),
```

Comment:
using contains string so that we are immune to any possible changes from `FormValidation.aggregate`